src/Symfony/Component/HttpKernel/Event/ResponseEvent.php

```diff
@@ -20,6 +24,32 @@
  *
  * @author Bernhard Schussek <bschussek@gmail.com>
  */
-class ResponseEvent extends FilterResponseEvent
+class ResponseEvent extends KernelEvent
 {
+    private $response;
+
+    public function __construct(HttpKernelInterface $kernel, Request $request, int $requestType, Response $response)
+    {
+        parent::__construct($kernel, $request, $requestType);
+
+        $this->setResponse($response);
+    }
+
+    /**
+     * Returns the current response object.
+     *
+     * @return Response
+     */
+    public function getResponse()
```

Comment:
let's use a real type instead? could require making either the class or the method final in 4.4